lib/api/controllers/controller.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const
+  {
+    errors: {
+      InternalError: KuzzleInternalError
+    }
+  } = require('kuzzle-common-objects');
+
+// Base class for all controllers (except the funnel one)
+class BaseController {
+  constructor(kuzzle) {
```

Comment:
Proposal: pass the `actions` list as constructor argument and do the `this.actions = new Set(actions)` in the BaseController constructor to avoid unnecessary code duplication

Reply:
Resolving: same as the previous comment.